### PR TITLE
Fix Half CompareTo on negatives

### LINKED ISSUE #49983: Issue : CompareTo of Half type when parameters are negative value
Environment : dotnet core 5.0

Problem is `CompareTo` of Half type when parameters are negative value.
I expect below example result.
```
Half half1 = (Half)(-180f);
Half half2 = (Half)(-180f);

var a1 = half1.Equals(half2); // true
var a2 = half1 == half2; // true
var a3 = half1.CompareTo(half2); // not 0, but -1
```
Why `a3` value is -1? not 0 ?

I follow dotnet core code and find out some strange code.
`Half operator <` that called in `CompareTo` is right   ?
```
public static bool operator <(Half left, Half right)
{
    if (IsNaN(left) || IsNaN(right))
    {
        // IEEE defines that NaN is unordered with respect to everything, including itself.
        return false;
    }

    bool leftIsNegative = IsNegative(left);

    if (leftIsNegative != IsNegative(right))
    {
        // When the signs of left and right differ, we know that left is less than right if it is
        // the negative value. The exception to this is if both values are zero, in which case IEEE
        // says they should be equal, even if the signs differ.
        return leftIsNegative && !AreZero(left, right);
    }
    return (left._value < right._value) ^ leftIsNegative;
}
```
When both left and right is same value of negative, `<` make `-1` to return value.

Thanks to read this issue. Somebody help me :cry:

### PR DESCRIPTION
**Pull request by ntovas**
fix #49983

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Half.cs [hunk 1]
```diff
@@ -385,6 +385,11 @@ public int CompareTo(object? obj)
         /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
         public int CompareTo(Half other)
         {
+            if (this == other)
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Half.cs:388]: It's not clear why this fix is correct.

`float` and `double` are both implemented as `<, >, ==, NaN` (how the code is prior to this PR), which leads me to believe that `<` or `>` are incorrectly handling an equals case.

Comment (jeffhandley) [reply src/libraries/System.Private.CoreLib/src/System/Half.cs:388]: @tannergooding What do you think we should do on this? I saw your comment above, but I wasn't sure if you were hoping @ntovas could share more info or if you had other ideas.

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Half.cs:388]: As is, I don't believe this is the correct fix. It needs someone (either the PR author, myself, or @pgovind) to do the further investigation as to where the actual bug is and where the `CompareTo` logic (which is identical to what `float`/`double` do) is failing.

Comment (jeffhandley) [reply src/libraries/System.Private.CoreLib/src/System/Half.cs:388]: @ntovas Do you think you'd be able to investigate this further?

**Comment by jeffhandley** [comment]
I'm closing this PR as we aren't confident in the fix. @ntovas please don't hesitate to open a new PR if you're able to address the feedback and questions posted here. Thank you!

**Pull request closed without merging**